strawberry/extensions/directives.py

```diff
@@ -14,58 +18,69 @@
+            strawberry_directive, arguments = process_directive(
+                directive, value, root, info
             )
+            value = await await_maybe(strawberry_directive.resolver(**arguments))
 
-        return result
+        return value
 
 
 class DirectivesExtensionSync(Extension):
-    # TODO: we might need the graphql info here
-    def resolve(self, _next, root, info, *args, **kwargs) -> AwaitableOrValue[Any]:
-        result = _next(root, info, *args, **kwargs)
+    def resolve(
+        self, _next, root, info: GraphQLResolveInfo, *args, **kwargs
+    ) -> AwaitableOrValue[Any]:
+        value = _next(root, info, *args, **kwargs)
 
         for directive in info.field_nodes[0].directives:
-            directive_name = directive.name.value
-
-            if directive_name in SPECIFIED_DIRECTIVES:
+            if directive.name.value in SPECIFIED_DIRECTIVES:
                 continue
-
-            # TODO: support converting lists
-            arguments = {
-                argument.name.value: argument.value.value
-                for argument in directive.arguments
-            }
-
-            schema: Schema = info.schema._strawberry_schema
-            strawberry_directive = schema.get_directive_by_name(directive_name)
-            assert (
-                strawberry_directive is not None
-            ), f"Directive {directive_name} not found"
-
-            result = strawberry_directive.resolver(result, **arguments)
-
-        return result
+            strawberry_directive, arguments = process_directive(
+                directive, value, root, info
+            )
+            value = strawberry_directive.resolver(**arguments)
+
+        return value
+
+
+def process_directive(
+    directive: DirectiveNode,
+    value: Any,
+    root: Any,
+    info: GraphQLResolveInfo,
+) -> Tuple[StrawberryDirective, Dict[str, Any]]:
+    """Get a `StrawberryDirective` from ``directive` and prepare its arguments."""
+    directive_name = directive.name.value
+    schema: Schema = info.schema._strawberry_schema  # type: ignore
+
+    strawberry_directive = schema.get_directive_by_name(directive_name)
+    assert strawberry_directive is not None, f"Directive {directive_name} not found"
+
+    # TODO: support converting lists
+    arguments = {
+        argument.name.value: argument.value.value  # type: ignore
+        for argument in directive.arguments
+    }
+    resolver = strawberry_directive.resolver
+
+    info_parameter = resolver.info_parameter
+    root_parameter = resolver.root_parameter
+    value_parameter = resolver.value_parameter
+    if info_parameter:
+        field: StrawberryField = schema.get_field_for_type(  # type: ignore
+            field_name=info.field_name,
+            type_name=info.parent_type.name,
+        )
+        arguments[info_parameter.name] = Info(_raw_info=info, _field=field)
+    if root_parameter:
+        arguments[root_parameter.name] = root
```

Comment:
this is not needed here, right?

Reply:
after taking care of this we can merge :D

Reply:
Good spot! Fixed in [733455d](https://github.com/strawberry-graphql/strawberry/pull/1713/commits/733455daf0f0c289d1d07f1f7689126d4b5704a3) 😊